features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/domain/ResourceRequestFactory.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.domain
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.net.Request
+import com.datadog.android.api.net.RequestFactory
+import com.datadog.android.api.storage.RawBatchEvent
+import com.datadog.android.sessionreplay.internal.exception.InvalidPayloadFormatException
+import com.datadog.android.sessionreplay.internal.net.RawEventsToResourcesMapper
+import com.datadog.android.sessionreplay.internal.recorder.SessionReplayResource
+import okhttp3.RequestBody
+import okio.Buffer
+import java.io.IOException
+import java.io.ObjectInputStream
+import java.io.StreamCorruptedException
+import java.lang.NullPointerException
+import java.util.Locale
+import java.util.UUID
+
+internal class ResourceRequestFactory(
```

Comment:
if I understand correctly, this class is not used right now, but it will be used in the future with another feature?

Reply:
Yes, exactly. This will be connected to the ResourceFeature in the next pr